src/Build/Logging/BinaryLogger/BinaryLogReplayEventSource.cs

```diff
@@ -14,6 +15,14 @@ namespace Microsoft.Build.Logging
     /// <remarks>The class is public so that we can call it from MSBuild.exe when replaying a log file.</remarks>
     public sealed class BinaryLogReplayEventSource : EventArgsDispatcher
     {
+        /// Touches the <see cref="ItemGroupLoggingHelper"/> static constructor
+        /// to ensure it initializes <see cref="TaskParameterEventArgs.MessageGetter"/>
+        /// and <see cref="TaskParameterEventArgs.DictionaryFactory"/>
+        static BinaryLogReplayEventSource()
+        {
+            _ = ItemGroupLoggingHelper.ItemGroupIncludeLogMessagePrefix;
```

Comment:
Would the static constructor get called if somebody does a build with the inproc node turned off, with no binlogger, but using custom loggers that handle the new TaskParameterEventArgs? Or would they end up with the generic implementation. I wonder if it's worth touching the static constructor from the BuildManager as well.

Reply:
Devious scenario, very clever ;) I decided to touch the static constructor from LoggingService, which will likely achieve the same effect. If this is still not enough, then those exotic loggers are responsible for doing it themselves ;)